raft/src/main/java/org/apache/kafka/raft/RaftClient.java

```diff
@@ -172,15 +176,17 @@ default void beginShutdown() {}
      * uncommitted entries after observing an epoch change.
      *
      * @param epoch the current leader epoch
+     * @param requiredEndOffset if this is set, it is the offset we must use.
```

Comment:
nit "offset which must be exactly the end offset of the atomic append" or something more precise than "use"